app/components/forms/events/view/create-access-code.js

```diff
@@ -100,49 +89,47 @@ export default Component.extend(FormMixin, {
         },
         accessCodeEndDate: {
           identifier : 'end_date',
-          optional   : false,
+          optional   : true,
           rules      : [
-            {
-              type   : 'empty',
-              prompt : this.get('l10n').t('Please enter the proper date')
-            },
             {
               type   : 'date',
               prompt : this.get('l10n').t('Please enter the proper date')
             }
           ]
-        },
-        startTime: {
-          identifier : 'start_time',
-          depends    : 'start_date',
-          rules      : [
-            {
-              type   : 'empty',
-              prompt : this.get('l10n').t('Please give a start time')
-            }
-          ]
-        },
-        endTime: {
-          identifier : 'end_time',
-          depends    : 'end_date',
-          rules      : [
-            {
-              type   : 'empty',
-              prompt : this.get('l10n').t('Please give an end time')
-            }
-          ]
         }
       }
     };
   },
   accessCode : '',
   accessUrl  : computed('data.code', function() {
     const params = this.get('router._router.currentState.routerJsState.params');
-    return location.origin + this.get('router').urlFor('public', params['events.view'].event_id, { queryParams: { access_code: this.get('data.code') } });
+    this.set('data.accessUrl', location.origin + this.get('router').urlFor('public', params['events.view'].event_id, { queryParams: { access_code: this.get('data.code') } })) ;
+    return this.get('data.accessUrl');
   }),
   actions: {
-    submit() {
+    toggleAllSelection(allTicketTypesChecked) {
+      this.set('allTicketTypesChecked', allTicketTypesChecked);
+      if (allTicketTypesChecked) {
+        this.set('data.tickets', this.get('data.event.tickets').slice());
+      }
+    },
+    updateTicketSelections(newSelection) {
+      if (newSelection.length === this.get('data.event.tickets').length) {
+        this.set('allTicketTypesChecked', true);
+      } else {
+        this.set('allTicketTypesChecked', false);
+      }
+    },
+    submit(data) {
       this.onValid(() => {
+        data.save()
+          .then(() => {
+            this.get('notify').success(this.get('l10n').t('Access code has been successfully created.'));
+            this.get('router').transitionTo('events.view.tickets.access-codes');
```

Comment:
@CosmicCoder96 @niranjan94 Need your help here. When I redirect the user back to the access-codes list after creation of the access-codes, the list isn't updated. I need to reload the page in order for the list to be updated. 
How to resolve this ?